### PR TITLE
[SPARK-28803][DOCS][SQL] Document DESCRIBE TABLE in SQL Reference

### PR DESCRIPTION
**Pull request by dilipbiswal**
### What changes were proposed in this pull request?
Document DESCRIBE TABLE statement in SQL Reference Guide. 

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

**Before:**
There was no documentation for this.

**After.**
Screen Shot 2019-08-31 at 1 53 35 PM
Screen Shot 2019-08-31 at 1 53 50 PM
Screen Shot 2019-08-31 at 1 54 02 PM
Screen Shot 2019-08-31 at 1 54 15 PM
Screen Shot 2019-08-31 at 1 54 30 PM
Screen Shot 2019-08-31 at 1 54 42 PM

### How was this patch tested?
Tested using jykyll build --serve

### CODE REVIEW

**Comment by dilipbiswal** [comment]
cc @gatorsmile

File: docs/sql-ref-syntax-aux-describe-table.md [hunk 1]
```diff
@@ -18,5 +18,53 @@ license: |
   See the License for the specific language governing permissions and
   limitations under the License.
 ---
+### Description
+DESCRIBE TABLE statement is used to return the metadata information of a
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-table.md]: Back-tick reserved words

File: docs/sql-ref-syntax-aux-describe-table.md [hunk 2]
```diff
@@ -18,5 +18,53 @@ license: |
...
+table. The basic metadata information includes column name, column type
+and column comment. If EXTENDED mode is specified then additional
+metadata such as parent database, owner, created and last access time etc.
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-table.md]: remove 'etc.'

File: docs/sql-ref-syntax-aux-describe-table.md [hunk 3]
```diff
@@ -18,5 +18,53 @@ license: |
...
+### Syntax
+{% highlight sql %}
+{DESC | DESCRIBE} [TABLE] [format] tableIdentifier [partition-spec] [desc-col]
+
+format:= EXTENDED | FORMATTED
+tableIdentifier:= [db.][table]
+parition-spec:= PARTITION (part_col1 = value1, part_col2 = value2, ...) 
+desc-col:= [db.][table.]column-name
+{% endhighlight %}
+**Note**<br>
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-table.md]: Do you need `<br>`? try adding blank lines instead

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-ref-syntax-aux-describe-table.md (hunk 3 above)
Comment (dilipbiswal) [reply docs/sql-ref-syntax-aux-describe-table.md]: @srowen No.. did not need it. Not sure why i put it in the first place. Thanks for catching it.

**Comment by gatorsmile** [comment]
LGTM 

Thanks! Merged to master.

**Pull request closed without merging**